sql/core/src/main/scala/org/apache/spark/sql/execution/command/CommandUtils.scala

```diff
@@ -62,6 +66,20 @@ object CommandUtils extends Logging {
       if (isNewStats) {
         val newStats = CatalogStatistics(sizeInBytes = newSize)
         catalog.alterTableStats(table.identifier, Some(newStats))
+
+        if (!isDropPartition && table.partitionColumnNames.nonEmpty) {
```

Comment:
I think here at line 64 we've already calculated size for each partition so it'd be nice if we can just re-use it.

Perhaps `CommandUtils.calculateTotalSize` can return a map from partition to its size, so that we can use the result and filter based on the input `partitionSpecs`.

Reply:
updated